apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -47,90 +44,66 @@ import { UpsertWorkflowCommand, UpsertWorkflowDataCommand } from './upsert-workf
 @Injectable()
 export class UpsertWorkflowUseCase {
   constructor(
-    private createWorkflowGenericUsecase: CreateWorkflowGeneric,
-    private updateWorkflowGenericUsecase: UpdateWorkflowGeneric,
+    private createWorkflowV0Usecase: CreateWorkflowV0Usecase,
+    private updateWorkflowV0Usecase: UpdateWorkflowV0Usecase,
     private notificationGroupRepository: NotificationGroupRepository,
     private getWorkflowByIdsUseCase: GetWorkflowByIdsUseCase,
     private getWorkflowUseCase: GetWorkflowUseCase,
     private buildStepIssuesUsecase: BuildStepIssuesUsecase,
     private controlValuesRepository: ControlValuesRepository,
     private upsertControlValuesUseCase: UpsertControlValuesUseCase,
-    private analyticsService: AnalyticsService,
-    private notificationTemplateRepository: NotificationTemplateRepository
+    private analyticsService: AnalyticsService
   ) {}
 
   @InstrumentUsecase()
   async execute(command: UpsertWorkflowCommand): Promise<WorkflowResponseDto> {
```

Comment:
The primary execute function's goal is to contain all the logic in one go without too much indirection. Moreover, the use case is symmetrical, one path for create, another path for update with identical function names.

Reply:
Love it!